src/graph.h

```diff
@@ -41,7 +41,13 @@ struct State;
 /// it's dirty, mtime, etc.
 struct Node {
   Node(const std::string& path, uint64_t slash_bits)
-      : path_(path), slash_bits_(slash_bits) {}
+      : path_(path), slash_bits_(slash_bits) {
+#ifdef _WIN32
+    path_lower_.resize(path_.size());
+    transform(path_.begin(), path_.end(),
+              path_lower_.begin(), ::tolower);
```

Comment:
::tolower() is locale-dependent and, as far as I know, is not recommended to compare file paths. What about using `LCMapStringEx` instead with `LOCAL_NAME_INVARIANT` and `LCMAP_UPPERCASE` ?

https://learn.microsoft.com/en-us/windows/win32/api/winnls/nf-winnls-lcmapstringa

https://learn.microsoft.com/en-us/archive/blogs/greggm/comparing-file-names-in-native-code

Reply:
`LCMapStringEx` only seems available as a wide-character API.  This is a hot path so I'd like to avoid the cost of extra conversion buffers.  I moved the implementation to a dedicated helper and left a FIXME comment to consider this in the future.  Note that `src/disk_interface.cc` already uses `tolower`.